pymeasure/instruments/hp/hp4145x.py

```diff
@@ -0,0 +1,1116 @@
+    def voltage_name(self, value):
+        if len(value.strip()) > 6 or len(value.strip()) < 1:
+            raise ValueError("Name must be less than six and at least one characters")
+
+        self._voltage_name = value.upper()
+        self._disabled = False
+
+        if not self.manual_flush:
+            self.write("DE VM %d, '%s'" % (self.id, self._voltage_name))
+            self.check_errors()
+
+    def flush_channel_definition(self):
+        """
+        Flushes the channel definition manually in case the :attr:`manual_flush` option is set.
+        """
+        if not self._disabled:
+            self.write("DE VM %d, '%s'" % (self.id, self._voltage_name))
+            self.check_errors()
+
+
+class SMU(Channel):
+    _voltage_range = 0
+    _voltage_ranges = {0: 0, 20: 1, 40: 2, 100: 3}
+    _current_range = 0
+    _current_ranges = {0: 0, 1e-9: 1, 10e-9: 2, 100e-9: 3,
+                       1e-6: 4, 10e-6: 5, 100e-6: 6, 1e-3: 7, 10e-3: 8, 100e-3: 9}
+
+    _current_max = 100e-3
+    _current_compliance = 0.0
+
+    _voltage_max = 100
+    _voltage_compliance = 0.0
+
+    _voltage_name = ''
+    _current_name = ''
+
+    _channel_function = 3
+    _channel_functions = {"VAR1": 1, "VAR2": 2, "CONST": 3, "VAR1'": 4}
+
+    _channel_mode = 3
+    _channel_modes = {'V': 1, 'I': 2, 'COM': 3}
+
+    _voltage = 0
+    _current = 0
+
+    _disabled = False
+
+    manual_flush = False
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._voltage_name = f"V{self.id}"
+        self._current_name = f"I{self.id}"
+
+    @property
+    def voltage(self):
+        """
+        Controls the output voltage of on of the SMU channels.
+        Automatically puts the SMU into force voltage mode once set.
+
+        .. code-block:: python
+
+            hp4145a.SMU2.current_compliance = 10e-3
+            hp4145a.SMU2.voltage = 10
+
+        """
+        if self.parent.mode == 'USER_MODE':
+            value = self.ask("TV {ch}")
+        else:
+            raise LookupError("Property can't be read in 'SYSTEM_MODE'")
+
+        self.check_get_errors()
+        return value
+
+    @voltage.setter
+    def voltage(self, value):
+        max_voltage = 100 if self._voltage_range == 0 else self._voltage_ranges[self._voltage_range]
+        value = strict_range(value, [0, max_voltage])
+
+        if self.parent.mode == 'USER_MODE':
+            self.write("DV {ch}, %d, %f, %f" %
+                       (self._voltage_range, value, self._current_compliance))
+        else:
+            # SYSTEM_MODE
+            if self._channel_function == self._channel_functions['CONST']:
+                self._voltage = value
+                if not self.manual_flush:
+                    self.flush_source_setup()
+            else:
+                raise RuntimeError("In SYSTEM_MODE only allowed with channel_function = 'CONST'")
+        self.check_set_errors()
+
+    @property
+    def current(self):
```

Comment:
see voltage

Reply:
That is a setpoint, as that is your desired current, right? It is not the current measured, is it?